install_sct.bat

```diff
@@ -75,64 +75,93 @@ if %SCT_INSTALL_TYPE%==in-place (
   set SCT_DIR=%USERPROFILE%\sct_%SCT_VERSION%
 )
 
-rem Allow user to set a custom installation directory
-:while_loop_sct_dir
+rem Validate the default installation directory 
+rem If it's not valid, don't propose it to the user
+call :validate_sct_dir >nul || goto :while_prompt_custom_path
+
+rem Count blank attempts when choosing a custom install path
+rem to avoid infinite loops
+set attempt=0
+
+:while_prompt_default_path
   echo:
   echo ### SCT will be installed here: [%SCT_DIR%]
-  set keep_default_path=yes
-  :while_loop_path_agreement
-    set /p keep_default_path="### Do you agree? [y]es/[n]o: "
-    echo %keep_default_path% | findstr /b [YyNn] >nul 2>&1 || goto :while_loop_path_agreement
-  :done_while_loop_path_agreement
-
-  echo %keep_default_path% | findstr /b [Yy] >nul 2>&1
-  if %errorlevel% EQU 0 (
-    rem user accepts default path, so exit loop
-    goto :done_while_loop_sct_dir
+
+  rem The non-interactive default is to accept
+  set keep_default_path=y
+  set /p keep_default_path="### Do you agree? [y]es/[n]o: "
+
+  rem Either the installation is non-interactive,
+  rem or the user accepts the default path
+  if /i ["%keep_default_path%"]==["y"] goto :done_sct_dir
+
+  rem The user wants a non-default path
+  if /i ["%keep_default_path%"]==["n"] goto :while_prompt_custom_path
+goto :while_prompt_default_path
+
+:while_prompt_custom_path
+  set /a attempt=attempt+1
+  if !attempts! GTR 10 (
+    rem The installation is probably non-interactive,
+    rem and the default path is invalid, so fail
+      echo ### ERROR: The default installation directory is invalid, and no input was detected.
+      echo            Please make sure to enter an input, or run the installer in interactive mode instead.
+      goto error

```

Comment:
I guess technically we can trip this condition by selecting `n` and then selecting an invalid path 10 times in a row.

While very rare, I guess it makes the message invalid. (I had already rewritten this message once, but I guess my understanding was off?)

```suggestion
    rem The install path was invalid 10 times, so this is probably non-interactive. Halt.
      echo ### ERROR: The chosen installation directory is invalid, and no valid input was passed.
      echo            Please make sure to enter a valid input.
      goto error

```